### ISSUE TITLE
Rename a file with / or other pathological chars causes heartburn 

### ISSUE THREAD

**Issue by michael-conway** [DICE-UNC/irods-cloud-browser#105]
for example, rename a data object like this...
![screenshot from 2015-09-15 16 05 07](https://cloud.githubusercontent.com/assets/1097261/9888542/aaa07c28-5bc3-11e5-877a-21e13dfc096e.png)

slashes in the rename file name are particularly a problem, try some other chars

**Comment by michael-conway** [comment]
This is weird, log of rename of openvpn.sh on test local server is currently hanging in rename dialog

log messages

```

2015-09-16 09:59:41,075 [http-bio-8080-exec-9] INFO  pub.CollectionListingUtils  - ObjStat [absolutePath=/tempZone/home/test1/openvpn.sh, objectPath=, objectType=DATA_OBJECT, dataId=45589, checksum=, ownerName=test1, ownerZone=tempZone, objSize=29, createdAt=Wed Sep 09 15:37:10 EDT 2015, modifiedAt=Wed Sep 09 15:37:10 EDT 2015, specColType=NORMAL, collectionPath=, cacheDir=, cacheDirty=false, replNumber=0, standInGeneratedObjStat=false]
2015-09-16 09:59:41,075 [http-bio-8080-exec-9] INFO  io.IRODSFileImpl  - paths different, and a file is being renamed
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] INFO  pub.IRODSFileSystemAOImpl  - renaming file:/tempZone/home/test1/openvpn.sh
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] INFO  pub.IRODSFileSystemAOImpl  -  to:/tempZone/home/test1/open/huh.sh
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] INFO  io.IRODSFileImpl  - isFile() for path:/tempZone/home/test1/openvpn.sh
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] INFO  pub.IRODSFileSystemAOImpl  - getObjStat(final String irodsAbsolutePath)
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] INFO  pub.CollectionAndDataObjectListAndSearchAOImpl  - retrieveObjectStatForPathWithHeuristicPathGuessing()
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] DEBUG pub.IRODSAccessObjectFactoryImpl  - creating access object factory
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] DEBUG connection.IRODSSession  - session using previously established connection:AbstractIRODSMidLevelProtocol [irodsConnection=irods://test1.tempZone@mikesubuntu:1247/http-bio-8080-exec-9/1442405501004, irodsProtocolManager=org.irods.jargon.core.connection.IRODSSimpleProtocolManager@713fdebb, irodsServerProperties=IRODS server properties
    icat enabled?:ICAT_ENABLED
    Server boot time:1442401920
    Rel version:rods4.2.0
    API version:d
    zone:tempZone
    eirods:true
, irodsSession=org.irods.jargon.core.connection.IRODSSession@4d7549ff, startupResponseData=StartupResponseData:
   status:0
   relVersion:rods4.2.0
   apiVersion:d
   reconnPort:20067
   reconnAddr:mikesubuntu
   cookie:256902774, forceSslFlush=false, log=org.slf4j.impl.GrailsLog4jLoggerAdapter(org.irods.jargon.core.connection.AbstractIRODSMidLevelProtocol), authResponse=org.irods.jargon.core.connection.auth.AuthResponse@65d023eb, irodsAccount=irods://test1@mikesubuntu:1247]
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - calling irods function with byte array
2015-09-16 09:59:41,076 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - calling irods function with:<DataObjInp_PI><objPath>/tempZone/home/test1/openvpn.sh</objPath>
<createMode>0</createMode>
<openFlags>0</openFlags>
<offset>0</offset>
<dataSize>0</dataSize>
<numThreads>0</numThreads>
<oprType>0</oprType>
<KeyValPair_PI><ssLen>0</ssLen>
</KeyValPair_PI>
</DataObjInp_PI>
2015-09-16 09:59:41,077 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - api number is:633
2015-09-16 09:59:41,077 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - functionID: 633
2015-09-16 09:59:41,077 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - header:<MsgHeader_PI><type>RODS_API_REQ</type><msgLen>274</msgLen><errorLen>0</errorLen><bsLen>0</bsLen><intInfo>633</intInfo></MsgHeader_PI>
2015-09-16 09:59:41,078 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - reading message from irods
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - IRODSConnection.read, byte array size =  4
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - offset = 0
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - length = 4
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - IRODSConnection.read, byte array size =  141
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - offset = 0
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - length = 141
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - message length:281
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - error length:0
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - bytesLength:0
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - info value:1
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - message length greater than zero
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - IRODSConnection.read, byte array size =  281
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - offset = 0
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - length = 281
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - message from IRODS read back:<RodsObjStat_PI><objSize>29</objSize>
<objType>1</objType>
<dataMode>33188</dataMode>
<dataId>45589</dataId>
<chksum></chksum>
<ownerName>test1</ownerName>
<ownerZone>tempZone</ownerZone>
<createTime>01441827430</createTime>
<modifyTime>01441827430</modifyTime>
</RodsObjStat_PI>
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - bytes length is not zero
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG pub.CollectionListingUtils  - response from objStat: <RodsObjStat_PI><objSize>29</objSize>
<objType>1</objType>
<dataMode>33188</dataMode>
<dataId>45589</dataId>
<chksum></chksum>
<ownerName>test1</ownerName>
<ownerZone>tempZone</ownerZone>
<createTime>01441827430</createTime>
<modifyTime>01441827430</modifyTime>
<MsgHeader_PI><type>RODS_API_REPLY</type>
<msgLen>281</msgLen>
<errorLen>0</errorLen>
<bsLen>0</bsLen>
<intInfo>1</intInfo>
</MsgHeader_PI>
</RodsObjStat_PI>
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] INFO  pub.CollectionListingUtils  - ObjStat [absolutePath=/tempZone/home/test1/openvpn.sh, objectPath=, objectType=DATA_OBJECT, dataId=45589, checksum=, ownerName=test1, ownerZone=tempZone, objSize=29, createdAt=Wed Sep 09 15:37:10 EDT 2015, modifiedAt=Wed Sep 09 15:37:10 EDT 2015, specColType=NORMAL, collectionPath=, cacheDir=, cacheDirty=false, replNumber=0, standInGeneratedObjStat=false]
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.IRODSSession  - session using previously established connection:AbstractIRODSMidLevelProtocol [irodsConnection=irods://test1.tempZone@mikesubuntu:1247/http-bio-8080-exec-9/1442405501004, irodsProtocolManager=org.irods.jargon.core.connection.IRODSSimpleProtocolManager@713fdebb, irodsServerProperties=IRODS server properties
    icat enabled?:ICAT_ENABLED
    Server boot time:1442401920
    Rel version:rods4.2.0
    API version:d
    zone:tempZone
    eirods:true
, irodsSession=org.irods.jargon.core.connection.IRODSSession@4d7549ff, startupResponseData=StartupResponseData:
   status:0
   relVersion:rods4.2.0
   apiVersion:d
   reconnPort:20067
   reconnAddr:mikesubuntu
   cookie:256902774, forceSslFlush=false, log=org.slf4j.impl.GrailsLog4jLoggerAdapter(org.irods.jargon.core.connection.AbstractIRODSMidLevelProtocol), authResponse=org.irods.jargon.core.connection.auth.AuthResponse@65d023eb, irodsAccount=irods://test1@mikesubuntu:1247]
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - calling irods function with byte array
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - calling irods function with:<DataObjCopyInp_PI><DataObjInp_PI><objPath>/tempZone/home/test1/openvpn.sh</objPath>
<createMode>0</createMode>
<openFlags>0</openFlags>
<offset>0</offset>
<dataSize>0</dataSize>
<numThreads>0</numThreads>
<oprType>11</oprType>
<KeyValPair_PI><ssLen>0</ssLen>
</KeyValPair_PI>
</DataObjInp_PI>
<DataObjInp_PI><objPath>/tempZone/home/test1/open/huh.sh</objPath>
<createMode>0</createMode>
<openFlags>0</openFlags>
<offset>0</offset>
<dataSize>0</dataSize>
<numThreads>0</numThreads>
<oprType>11</oprType>
<KeyValPair_PI><ssLen>0</ssLen>
</KeyValPair_PI>
</DataObjInp_PI>
</DataObjCopyInp_PI>
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.IRODSMidLevelProtocol  - api number is:627
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - functionID: 627
2015-09-16 09:59:41,079 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - header:<MsgHeader_PI><type>RODS_API_REQ</type><msgLen>591</msgLen><errorLen>0</errorLen><bsLen>0</bsLen><intInfo>627</intInfo></MsgHeader_PI>
2015-09-16 09:59:41,081 [http-bio-8080-exec-9] DEBUG connection.AbstractIRODSMidLevelProtocol  - reading message from irods
2015-09-16 09:59:41,081 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - IRODSConnection.read, byte array size =  4
2015-09-16 09:59:41,081 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - offset = 0
2015-09-16 09:59:41,081 [http-bio-8080-exec-9] DEBUG connection.AbstractConnection  - length = 4



```

**Comment by michael-conway** [comment]
Note the dataObjCopy is properly sent and blocking, waiting on a response,

iRODS log is curious

```
mconway@mikesubuntu:/var/lib/irods/iRODS/server/log$ tail -f rodsLog.2015.09.16
Sep 16 09:59:41 pid:23346 DEBUG1: cmlGetOneRowFromSqlBV select object_id from R_OBJT_ACCESS OA, R_DATA_MAIN DM, R_USER_GROUP UG, R_USER_MAIN UM, R_TOKN_MAIN TM where OA.object_id=? and UM.user_name=? and UM.zone_name=? and UM.user_type_name!='rodsgroup' and UM.user_id = UG.user_id and OA.object_id = DM.data_id and UG.group_user_id = OA.user_id and OA.access_type_id >= TM.token_id and  TM.token_namespace ='access_type' and TM.token_name = ? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: select object_id from R_OBJT_ACCESS OA, R_DATA_MAIN DM, R_USER_GROUP UG, R_USER_MAIN UM, R_TOKN_MAIN TM where OA.object_id=? and UM.user_name=? and UM.zone_name=? and UM.user_type_name!='rodsgroup' and UM.user_id = UG.user_id and OA.object_id = DM.data_id and UG.group_user_id = OA.user_id and OA.access_type_id >= TM.token_id and  TM.token_namespace ='access_type' and TM.token_name = ? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: cmlGetOneRowFromSqlBV select coll_id from R_DATA_MAIN DM, R_OBJT_ACCESS OA, R_USER_GROUP UG, R_USER_MAIN UM, R_TOKN_MAIN TM where DM.data_id=? and UM.user_name=? and UM.zone_name=? and UM.user_type_name!='rodsgroup' and UM.user_id = UG.user_id and OA.object_id = DM.data_id and UG.group_user_id = OA.user_id and OA.access_type_id >= TM.token_id and TM.token_namespace ='access_type' and TM.token_name = 'own' limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: select coll_id from R_DATA_MAIN DM, R_OBJT_ACCESS OA, R_USER_GROUP UG, R_USER_MAIN UM, R_TOKN_MAIN TM where DM.data_id=? and UM.user_name=? and UM.zone_name=? and UM.user_type_name!='rodsgroup' and UM.user_id = UG.user_id and OA.object_id = DM.data_id and UG.group_user_id = OA.user_id and OA.access_type_id >= TM.token_id and TM.token_namespace ='access_type' and TM.token_name = 'own' limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: cmlGetOneRowFromSqlBV select data_id from R_DATA_MAIN where data_name=? and coll_id=? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: select data_id from R_DATA_MAIN where data_name=? and coll_id=? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: cmlGetOneRowFromSqlBV select coll_id from R_COLL_MAIN where coll_name = ( select coll_name from R_COLL_MAIN where coll_id=? ) || ? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: select coll_id from R_COLL_MAIN where coll_name = ( select coll_name from R_COLL_MAIN where coll_id=? ) || ? limit 1
Sep 16 09:59:41 pid:23346 DEBUG1: begin
Sep 16 09:59:41 pid:23346 DEBUG1: update R_DATA_MAIN set data_name = ? where data_id=?





```


a PS of the process id shows


```
irods    23346  2129  0 08:11 ?        00:00:00 irodsAgent
postgres 23347  1347  0 08:11 ?        00:00:00 postgres: irods ICAT 127.0.0.1(58070) UPDATE waiting    

```

**Comment by michael-conway** [comment]
This is an irods issue, we may overhead in jargon...will keep parked here until that's sorted out
